### PR TITLE
handle exponent notation floats and _s

### LINKED ISSUE #103: Support _ thousands separator; handle exponent floating point notation
Go, JavaScript, and other languages support the following notation

1_000_000 for 1e6

BTW, is 1e6 supported?

### PR DESCRIPTION
**Pull request by ldemailly**
fixes #103 

doesn't complain about multiple consecutive `_` but... the float parser will in eval phase

### CODE REVIEW

File: lexer/lexer_test.go [hunk 1]
```diff
@@ -256,3 +256,72 @@ func TestNextTokenCommentEOLMode(t *testing.T) {
...
+		{".a", "."}, // not a valid float number yet we consume and it'll fail to convert later.
+		{"123.", "123."},
+		{".5", ".5"},
+		{"100.56", "100.56"},
+		{"1e3", "1e3"},
+		{"1.23e-3", "1.23e-3"},
+		{"1.23E+3", "1.23E+3"},
+		{"1e-3", "1e-3"},
+		{"12..3", "12."},
+		{"1e+3", "1e+3"},
```
Comment (ccoVeille) [inline lexer/lexer_test.go:275]: Please add 1e+1_000 to test 1e+1000

File: lexer/lexer_test.go [hunk 2]
```diff
@@ -256,3 +256,72 @@ func TestNextTokenCommentEOLMode(t *testing.T) {
...
+	tests := []struct {
+		input    string
+		expected string
+	}{
+		{".", "."},  // not a valid float number yet we consume and it'll fail to convert later.
+		{".a", "."}, // not a valid float number yet we consume and it'll fail to convert later.
+		{"123.", "123."},
+		{".5", ".5"},
+		{"100.56", "100.56"},
+		{"1e3", "1e3"},
```
Comment (ccoVeille) [inline lexer/lexer_test.go:270]: Please add 1e1_000 to test 1e1000

Comment (ldemailly) [reply lexer/lexer_test.go:270]: max exponent on float64 is 308 yet sure I can put more, the lexer won't mind (1e1_000 is no different than 1e1_2_3)

File: lexer/lexer_test.go (hunk 1 above)
Comment (ldemailly) [reply lexer/lexer_test.go:275]: done

File: lexer/lexer_test.go (hunk 2 above)
Comment (ldemailly) [reply lexer/lexer_test.go:270]: done nevertheless

**Comment by ldemailly** [comment]
this one is the ez one :) well though it took me a bit to get the lexer corner cases to work, but #102 is where additional eyes could be useful

**Pull request merged**